plasma.lua

```diff
@@ -1,127 +1,176 @@
--- plasma!
--- 
--- e1    time
--- k1+e1 func
--- e2    a
--- e3    b
--- k1+e2 c
--- k1+e3 d
---
--- submit new funcs!
-
 abs = math.abs
 floor = math.floor
 sin = math.sin
 cos = math.cos
 sqrt = math.sqrt
 
+--- @type function[]
+--- Array of functions to create patterns
```

Comment:
Like the rest of this code, there is plenty of this Lua documentation, which can actually be seen in the intended way while using the [Lua Language Server](https://github.com/LuaLS/lua-language-server) extension on [Visual Studio Code](https://code.visualstudio.com/).